samigo/samigo-impl/src/java/org/sakaiproject/samigo/impl/SamigoETSProviderImpl.java

```diff
@@ -0,0 +1,441 @@
+            replacementValues.put("assessmentTitle"     , pubAssFac.getTitle());
+            replacementValues.put("assessmentDueDate"   , pubAssFac.getDueDate() == null ? "" : pubAssFac.getDueDate().toString());
+            replacementValues.put("assessmentGradingID" , notificationValues.get("assessmentGradingID").toString());
+            replacementValues.put("submissionDate"      , notificationValues.get("submissionDate").toString());
+
+
+            RenderedTemplate rt                 = getRenderedTemplate(SamigoConstants.EMAIL_TEMPLATE_ASSESSMENT_SUBMITTED , user, replacementValues);;
+            // Assume assessmentSubmittedType is 1 to ensure rt is initialized
+            if (assessmentSubmittedType == 2) {
+                rt = getRenderedTemplate(SamigoConstants.EMAIL_TEMPLATE_ASSESSMENT_AUTO_SUBMITTED   , user, replacementValues);
+            } else if (assessmentSubmittedType == 3){
+                rt = getRenderedTemplate(SamigoConstants.EMAIL_TEMPLATE_ASSESSMENT_TIMED_SUBMITTED  , user, replacementValues);
+            }
+
+            String body = getBody(rt);
+
+            notifyStudent(user, rt, body, priStr);
+            notifyInstructor(siteID, pubAssFac.getInstructorNotification(), rt, body, priStr);
+        } catch(UserNotDefinedException e){
+            log.warn("UserNotDefined: " + notificationValues.get("userID").toString() );
+        }
+    }
+
+    private     void                notifyInstructor                (String siteID, Integer instructNoti , RenderedTemplate rt, String message,String priStr){
+        log.debug("notifyInstructor");
+
+        String              instructorRole          = serverConfigurationService.getString("samigo.instructorRole");
+        Map<User, Integer>  validUsers              = new HashMap<>();
+
+        if(instructorRole.equals("")){
+            instructorRole = "Instructor";
+        }
+
+        try{
+            Site            site                    = siteService.getSite(siteID);
+            Set<String>     siteUsersHasRole        = site.getUsersHasRole(instructorRole);
+
+            if( siteUsersHasRole.isEmpty()){ // try default "instructor" role
+                siteUsersHasRole = site.getUsersHasRole("maintain");
+            }
+
+            for(String userString : siteUsersHasRole){
+                try{
+                    if(!userString.equals(ADMIN)) {
+                        User user = userDirectoryService.getUser(userString);
+
+                        Integer uPref = getUserPreferences(user, priStr);
+                        validUsers.put(user, uPref);
+                    }
+                } catch(UserNotDefinedException e){
+                    log.warn("Instructor '" + userString +"' not found.");
+                }
+            }
+        } catch(org.sakaiproject.exception.IdUnusedException e){
+            //Site not found
+            log.warn("Site '" + siteID + "' not found.");
```

Comment:
Little more detail in the warning would be nice... i.e.

Site + siteID + not found while sending notifications for samigo auto submit job